packages/arb-token-bridge-ui/src/hooks/useDeposits.ts

```diff
@@ -0,0 +1,68 @@
+import useSWRImmutable from 'swr/immutable'
+import {
+  fetchDeposits,
+  FetchDepositParams,
+  Transaction
+} from 'token-bridge-sdk'
+import { PageParams } from '../components/TransactionHistory/TransactionsTable/TransactionsTable'
+import { useAppState } from '../state'
+import { MergedTransaction } from '../state/app/state'
+import { isPending, transformDeposits } from '../state/app/utils'
+import { useNetworksAndSigners } from './useNetworksAndSigners'
+
+export type CompleteDepositData = {
+  deposits: Transaction[]
+  pendingDeposits: Transaction[]
+  transformedDeposits: MergedTransaction[]
+}
+
+export const fetchCompleteDepositData = async (
+  depositParams: FetchDepositParams
+): Promise<CompleteDepositData> => {
+  // get the original deposits
+  const deposits = await fetchDeposits(depositParams)
+
+  // filter out pending deposits
+  const pendingDepositsMap = new Map<string, boolean>()
+  // get their complete transformed data (so that we get their exact status)
+  const completeDepositData = transformDeposits(deposits)
+  completeDepositData.forEach(completeTxData => {
+    if (isPending(completeTxData)) {
+      pendingDepositsMap.set(completeTxData.txId, true)
+    }
+  })
+  const pendingDeposits = deposits.filter(tx => pendingDepositsMap.get(tx.txID))
```

Comment:
Let's be a bit more explicit here that we're filtering to only elements that can be found inside the map
```suggestion
  const pendingDeposits = deposits.filter(tx => typeof pendingDepositsMap.get(tx.txID) !== 'undefined')
```